cpp-client/deephaven/dhclient/src/utility/table_maker.cc

```diff
@@ -1,43 +1,49 @@
 /*
  * Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
  */
-#include "deephaven/client/flight.h"
 #include "deephaven/client/utility/table_maker.h"
+
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "deephaven/client/flight.h"
 #include "deephaven/client/utility/arrow_util.h"
 #include "deephaven/dhcore/utility/utility.h"
+#include "deephaven/third_party/fmt/core.h"
 #include "deephaven/third_party/fmt/format.h"
 
+#include "arrow/array/array_base.h"
+#include "arrow/array/builder_nested.h"
+
+using deephaven::dhcore::utility::MakeReservedVector;
 using deephaven::client::TableHandle;
-using deephaven::client::utility::OkOrThrow;
-using deephaven::client::utility::ValueOrThrow;
 
 #include <memory>
 
 namespace deephaven::client::utility {
 TableMaker::TableMaker() = default;
 TableMaker::~TableMaker() = default;
 
-void TableMaker::FinishAddColumn(std::string name, internal::TypeConverter info) {
-  auto kv_metadata = std::make_shared<arrow::KeyValueMetadata>();
-  OkOrThrow(DEEPHAVEN_LOCATION_EXPR(kv_metadata->Set("deephaven:type", info.DeephavenType())));
-
-  auto field = std::make_shared<arrow::Field>(std::move(name), std::move(info.DataType()), true,
-      std::move(kv_metadata));
-  OkOrThrow(DEEPHAVEN_LOCATION_EXPR(schemaBuilder_.AddField(field)));
-
-  if (columns_.empty()) {
-    numRows_ = info.Column()->length();
-  } else if (numRows_ != info.Column()->length()) {
-    throw std::runtime_error(DEEPHAVEN_LOCATION_STR(
-        fmt::format("Column sizes not consistent: expected {}, have {}", numRows_,
-            info.Column()->length())));
+void TableMaker::FinishAddColumn(std::string name, std::shared_ptr<arrow::Array> data,
+    std::string deephaven_server_type_name) {
+  if (!column_infos_.empty()) {
```

Comment:
Depending on how many arguments you end up having to pass, this if block that looks to be doing some validation may be easier to read as a separate small function called `validate_X` or `check_X` but, your call, the whole function is short already so not a strong suggestion, more like an idea to think about

Reply:
Thanks. This validation has been moved to a later point, in its own method called `ValidateSchema()`. PTAL